### PR TITLE
completions/lxc: parse container names with numbers and other commands

### PR DESCRIPTION
**Pull request by pkasemir**
## Description

Hostnames are allowed to have number digits, so I've added numbers to the container name parsing function.

There are a few more commands available and which use container names as arguments, so I've added those also.

Fixes issue #

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -55,6 +55,8 @@ Completions
   - brightnessctl (:issue:`8758`)
   - rclone (:issue:`8819`)
   - tuned-adm (:issue:`8760`)
+- Improved completions for:
+  - lxc
```
Comment (krobelus) [inline CHANGELOG.rst:59]: thanks for remembering the changelog but we generally don't want to mention updates to existing completions

File: share/completions/lxc.fish [hunk 2]
```diff
@@ -41,11 +44,8 @@ complete --condition __fish_lxc_no_subcommand --command lxc --no-files --argumen
...
 complete --condition '__fish_seen_subcommand_from config' --command lxc --no-files --arguments "device get set unset show edit trust"
 
-# exec
-complete --condition '__fish_seen_subcommand_from exec' --command lxc --no-files --arguments "(__fish_lxc_list_containers)"
-
-# start
-complete --condition '__fish_seen_subcommand_from start' --command lxc --no-files --arguments "(__fish_lxc_list_containers)"
+# args taking container name
+for cmd in console copy delete exec export info move pause publish rename restart restore shell snapshot start stop
+	complete --condition "__fish_seen_subcommand_from $cmd" --command lxc --no-files --arguments "(__fish_lxc_list_containers)"
```
Comment (krobelus) [inline share/completions/lxc.fish:49]: __fish_seen_subcommand_from can take multiple potential subcommands, so how about

```fish
set -l subcommands_taking_name console copy delete exec export info move pause publish rename restart restore shell snapshot start stop
complete --condition "__fish_seen_subcommand_from $subcommands_taking_name" --command lxc --no-files --arguments "(__fish_lxc_list_containers)"
```

**Comment by krobelus** [comment]
Thanks!

File: CHANGELOG.rst (hunk 1 above)
Comment (zanchey) [reply CHANGELOG.rst]: I stopped adding them because it became overwhelming!

**Pull request merged**